File: server/src/main/com/broll/gainea/server/core/cards/impl/play/C_TeleportDesert.java
package com.broll.gainea.server.core.cards.impl.play;

import com.broll.gainea.server.core.cards.TeleportCard;
import com.broll.gainea.server.core.map.AreaType;
import com.broll.gainea.server.core.map.Location;
import com.broll.gainea.server.core.utils.LocationUtils;

import java.util.List;
import java.util.stream.Collectors;

public class C_TeleportDesert extends TeleportCard {
    public C_TeleportDesert() {
        super(48, "Wüstenkarawane", "Bewegt eine Truppe zu einer beliebigen Wüste auf der gleichen Karte");
    }

    @Override
    public List<Location> getTeleportTargets(Location from) {
        return LocationUtils.filterByType(from.getContainer().getExpansion().getAllLocations(), AreaType.DESERT).collect(Collectors.toList());
    }
}

File: server/src/main/com/broll/gainea/server/core/cards/impl/play/C_TeleportDesert.kt
package com.broll.gainea.server.core.cards.impl.play

import com.broll.gainea.server.core.Game
import com.broll.gainea.server.core.cards.TeleportCard
import com.broll.gainea.server.core.map.AreaType.DESERT
import com.broll.gainea.server.core.map.Location

class C_TeleportDesert : TeleportCard(48, "Wüstenkarawane", "Bewegt eine Truppe zu einer beliebigen Wüste auf der gleichen Karte") {
    override fun getTeleportTargets(from: Location) = from.container.expansion.allAreas.filter { it.type == DESERT }

    override fun validFor(game: Game) = game.map.allAreas.any { it.type == DESERT }
}
